autosklearn/pipeline/components/data_preprocessing/imputation/categorical_imputation.py

```diff
@@ -14,15 +21,31 @@ def __init__(self, random_state=None):
     def fit(self, X, y=None):
         import sklearn.impute
 
+        fill_value = None
+        if hasattr(X, 'columns'):
+            kind = X[X.columns[-1]].dtype.kind
+        else:
+            # Series, sparse and numpy have dtype
+            # Only DataFrame does not
+            kind = X.dtype.kind
+        if kind in ("i", "u", "f"):
+            # We do not want to impute a category with the default
+            # value (0 is the default) in case such default is in the
+            # train data already!
+            fill_value = 0
+            unique = np.unique(X)
+            while fill_value in unique:
+                fill_value -= 1
```

Comment:
Can we put the imputation after the ordinal encoding? If yes, we don't need this additional check for with which value to impute the data as we know that the data is always positive.

Reply:
Yes, that will be ideal, but I don't think this is yet supported:
```
enc = OrdinalEncoder()
>>> X = [['Male', 1], ['Female', 3], ['Female', 2], [np.nan, np.nan]]
>>> enc.fit(X)
Traceback (most recent call last):
...
ValueError: Input contains NaN
```

When [this](https://github.com/scikit-learn/scikit-learn/issues/11997) is addressed we can do that I think.